buildtest/cli/inspect.py

```diff
@@ -88,65 +94,118 @@ def fetch_test_names(report, names):
+
+
+def inspect_list(
+    report, terse=None, no_header=None, builder=None, color=None, pager=None
+):
     """This method list an output of test id, name, and buildspec file from the report cache. The default
     behavior is to display output in table format though this can be changed with terse format which will
     display in parseable format. This method implements command ``buildtest inspect list``
 
     Args:
         report (str): Path to report file
         terse (bool, optional): Print output in terse format
-        header (bool, optional): Determine whether to print header in terse format.
+        no_header (bool, optional): Determine whether to print header in terse format.
         builder (bool, optional): Print output in builder format which can be done via ``buildtest inspect list --builder``
         color (bool, optional): Print table output of ``buildtest inspect list`` with selected color
-
+        pager (bool, optional): Print output in paging format
     """
     consoleColor = checkColor(color)
 
     test_ids = report._testid_lookup()
 
     # implement command 'buildtest inspect list --builder'
     if builder:
-        builders = report.builder_names()
-        for name in builders:
-            print(name)
+        if pager:
+            with console.pager():
+                print_builders(report)
+            return
+
+        print_builders(report)
         return
 
+    table = {
+        "id": [],
+        "name": [],
+        "buildspec": [],
+    }
+
+    for identifier in test_ids.keys():
+        table["id"].append(identifier)
+        table["name"].append(test_ids[identifier]["name"])
+        table["buildspec"].append(test_ids[identifier]["buildspec"])
+
     # print output in terse format
     if terse:
-        # print column headers if --no-header is not specified
-        if not header:
-            console.print("[blue]id|name|buildspec")
-
-        for identifier in test_ids.keys():
-            console.print(
-                f"{identifier}|{test_ids[identifier]['name']}|{test_ids[identifier]['buildspec']}"
-            )
+        if pager:
+            with console.pager():
+                print_terse(table, no_header, consoleColor)
+            return
 
+        print_terse(table, no_header, consoleColor)
         return
 
-    table = Table(
-        "[blue]id",
-        "[blue]name",
-        Column(header="[blue]buildspec", overflow="fold"),
-        title="Test Summary by name, id, buildspec",
+    inspect_table = Table(
+        header_style="blue",
+        title="Test Summary by id, name, buildspec",
         row_styles=[consoleColor],
     )
-    for identifier in test_ids.keys():
-        table.add_row(
-            f"{identifier}",
-            f"{test_ids[identifier]['name']}",
-            f"{test_ids[identifier]['buildspec']}",
-        )
-    console.print(table)
+    inspect_table.add_column("id")
+    inspect_table.add_column("name")
+    inspect_table.add_column("buildspec")
```

Comment:
could you rewrite this as follows: 

```python
for column in table.keys():
    inspect_table.add_column(column)
```

I think both statements are equivalent but less code and also makes sure table columns are added based on keys in `table` variable